enterprise_access/apps/subsidy_request/models.py

```diff
@@ -306,3 +306,71 @@ def update_course_info_for_subsidy_request(sender, **kwargs):  # pylint: disable
         subsidy_type,
         str(subsidy_request.uuid),
     )
+
+
+class LearnerCreditRequest(SubsidyRequest):
+    """
+    Stores information related to a learner credit request.
+
+    .. no_pii: This model has no PII
+    """
+
+    subsidy_access_policy = models.ForeignKey(
+        'subsidy_access_policy.SubsidyAccessPolicy',
+        related_name="learner_credit_requests",
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        help_text="The SubsidyAccessPolicy that this learner credit request is associated with.",
+    )
+
+    assignment = models.OneToOneField(
+        LearnerContentAssignment,
+        related_name="credit_request",
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        help_text="The content assignment associated with this request."
+    )
+
+    history = HistoricalRecords()
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(
+                fields=['user', 'enterprise_customer_uuid', 'course_id'],
+                name='unique_learner_course_request',
+                condition=models.Q(state__in=[SubsidyRequestStates.REQUESTED, SubsidyRequestStates.PENDING]),
```

Comment:
The UniqueConstraint in LearnerCreditRequest ensures that only one active request (REQUESTED or PENDING state) can exist for a combination of user, enterprise_customer_uuid, and course_id.

Reply:
Question: Shouldn't the uniqueness be defined at a budget level instead of customer?